core/src/wallet/ethereum/EthereumLikeAccount.cpp

```diff
@@ -633,6 +633,21 @@ namespace ledger {
             getERC20Balance(erc20Address).callback(getContext(), callback);
         }
 
+        Future<std::vector<std::shared_ptr<api::BigInt>>> EthereumLikeAccount::getERC20Balances(const std::vector<std::string> &erc20Addresses) {
+            return _explorer->getERC20Balances(_keychain->getAddress()->toEIP55(), erc20Addresses)
```

Comment:
Ah woah, the explorers give us the history? That’s _cheating_!

Reply:
Actually it's just batching the calls of ERC20 balances